### ISSUE TITLE
Tailwind v1.0 compatibility

### ISSUE THREAD

**Issue by Shaderpixel** [bradlc/babel-plugin-tailwind-components#20]
Hi @bradlc, great work on creating this babel plugin. I am building my website using Gatsby and following the starter demo for Gatsby + Emotion + Tailwind (https://github.com/jlengstorf/gatsby-tailwind-demo). It works great with Tailwind v0.7.4. 

I was poking around Tailwind site and noticed that Adam is planning a release of v1.0 soon and there are some changes in the way the config files are setup (especially in the theme area). I tried to use the [beta v3 release](https://github.com/tailwindcss/tailwindcss/releases) with your plugin and run into an issue when I try to call the new color classes e.g. `bg-gray-300` I get  a `TypeError: Cannot read property 'gray-300' of undefined`. I am guessing this is related to how the config file structure has changed. I could be wrong but would love your input on this. Thanks.

**Comment by davecalnan** [comment]
Hi @Shaderpixel, I ran into the same issue and looked into what would be required for Tailwind 1.0 capability. Unfortunately it doesn't seem to be a trivial update.

## My exploration into the source code of this package and [tailwindcss](https://github.com/tailwindcss/tailwindcss/tree/next)

Because pre-1.0 installations have the entire config explicitly listed in the tailwind.js file, it's relatively straightforward to pull in that config and figure out all of the generated classes. (It seems a whole other thing to parse all of the CSS-in-JS and replace the classes with the correct styles).

However with new 1.0 configs, there is new work involved in taking the (likely) minimal tailwind.config.js file and [merging](https://github.com/tailwindcss/tailwindcss/blob/75380981106519d10dcec8c5f26c393bf96f5415/src/index.js#L42) it with the default Tailwind config.

The [default config](https://github.com/tailwindcss/tailwindcss/blob/next/stubs/defaultConfig.stub.js) has [several](https://github.com/tailwindcss/tailwindcss/blob/75380981106519d10dcec8c5f26c393bf96f5415/stubs/defaultConfig.stub.js#L214) [keys](https://github.com/tailwindcss/tailwindcss/blob/75380981106519d10dcec8c5f26c393bf96f5415/stubs/defaultConfig.stub.js#L215) that get a theme function which is [called within postcss](https://github.com/tailwindcss/tailwindcss/blob/next/src/lib/evaluateTailwindFunctions.js) when building the Tailwind classes.

All this is to say that the work involved to get the resultant config has gone from importing your tailwind.js config file to importing the new tailwind.config.js file, merging it with the default config, and then running all of the functions that get the theme passed to it.

## I'm going to try and update to support Tailwind 1.0

I love this package and would love to continue using it with Tailwind 1.*. I'm going to fork the package and get it 1.0 compatible and then create a pull request with the updates. @bradlc if you think that would be helpful, I would really appreciate your input! Thanks.

**Comment by bradlc** [comment]
Hey @Shaderpixel and @davecalnan 

v1 support is very nearly done on the [`next` branch](https://github.com/bradlc/babel-plugin-tailwind-components/tree/next). The main thing left to do is to update the static styles. I will try to publish it to npm this week under the `next` tag.

**Comment by DevanB** [comment]
@bradlc I, too, am curious about 1.0. Great work updating this! I'm curious to try it out once you release it to the `next` tag!

**Comment by bradlc** [comment]
Still got a bit to do but here's an initial version for anyone wanting to try it out 🎉 https://github.com/bradlc/babel-plugin-tailwind-components/releases/tag/v1.0.0-alpha.2

**Comment by Shaderpixel** [comment]
Awesome @bradlc, thanks for working so quickly on it. I will give it a whirl hopefully this weekend.

**Comment by pradel** [comment]
I tried and for now it works well!
Thank you so much @bradlc 🙏 
I tried to make it work with `styled-components` but no luck, I couldn't find how to make the custom config work.

**Comment by bradlc** [comment]
@pradel the config goes in a [`babel-plugin-macros` config file](https://github.com/kentcdodds/babel-plugin-macros/blob/master/other/docs/user.md#config-experimental). For `styled-components` you would want:

```js
module.exports = {
  tailwind: {
    styled: 'styled-components'
  }
}
```

Is that what you have?

**Comment by pradel** [comment]
@bradlc my bad, I was trying to set the config in `.babelrc` 🤦‍♂️ it's working now, thank you!

**Comment by Shaderpixel** [comment]
Hey @bradlc, I installed the tailwind.macro and using it with Gatsby. Here are my steps, please let me know if I am doing something wrong: I imported the macro into the document that I am working on and instead of using `styled` from `@emotion/styled` I will just use the `tw` tagged templates? 

One more question, is the following no longer doable or I am just doing it wrong. Any suggestions how to accomplish something like the following?
```JavaScript
const Paragraph = tw.p`
  flex justify-center items-center bg-gray-500
  color: ${props => (props.primary ? 'hotpink' : 'turquoise')};
`;
```

**Comment by bradlc** [comment]
Hey @Shaderpixel 

Yeah so instead of this (which still works):

```js
import styled from '@emotion/styled'
import tw from 'tailwind.macro'

const Paragraph = styled.p`
  ${tw`flex justify-center items-center bg-gray-500`}
`;
```

...you can do this:

```js
import tw from 'tailwind.macro'

const Paragraph = tw.p`flex justify-center items-center bg-gray-500`;
```

Interpolations like in your example are not supported right now. You would have to do something like:

```js
import styled from '@emotion/styled'
import tw from 'tailwind.macro'

const Paragraph = styled.p`
  ${tw`flex justify-center items-center bg-gray-500`}
  color: ${props => (props.primary ? 'hotpink' : 'turquoise')};
`;
```

In #22 I have an example of what I'd like to support with regards to interpolations. This would allow:

```js
const Paragraph = tw.p`
  flex justify-center items-center bg-gray-500
  ${props => (props.primary ? { color: 'hotpink' } : { color: 'turquoise' })};
`;
```

Or if you have those colors in your Tailwind config:

```js
const Paragraph = tw.p`
  flex justify-center items-center bg-gray-500
  ${props => (props.primary ? tw`text-hotpink` : tw`text-turquoise`)};
`;
```

Are you having trouble getting it to work at all right now?

**Comment by bradlc** [comment]
I have put together a PR for @jlengstorf's Gatsby demo which uses the alpha release: https://github.com/jlengstorf/gatsby-tailwind-demo/pull/8

**Comment by Shaderpixel** [comment]
Sweet! Thanks for all the examples and ultra fast response. My issue was I was testing with the old configuration inside of `gatsby-node.js` that uses the Babel plugin but with the new `tailwind.config.js` and it obviously error out 🤦‍♂️ . But the interpolation now works well and I am looking forward to using the new interpolation methods in the future. Thanks for all your hard work. Should I close this issue now since support for v1.0 is almost done or you can close it with your v1.0 release?

**Comment by bradlc** [comment]
No problem! Let me know if you run into any other issues, or have any questions/suggestions.

Let's leave this open until it's released properly :)

**Comment by ockam** [comment]
@bradlc I’m a bit confused by your gatsby-tailwind-demo PR, i.e. you are not using `babel-plugin-macros` in the packages. Is it required or not?

**Comment by bradlc** [comment]
@ockam Gatsby [supports `babel-plugin-macros` out-of-the-box](https://www.gatsbyjs.org/docs/babel-plugin-macros/) 🙌

**Comment by ockam** [comment]
OK. When using the following code, I get the proper html node but css class do not work (I’m using Emotion):
```
import tw from "tailwind.macro"
const Test = tw.p` bg-pink text-red`
```
I don’t see any special config in the example. What am I missing?

**Comment by bradlc** [comment]
Are you using the [default Tailwind v1 config](https://github.com/tailwindcss/tailwindcss/blob/next/stubs/defaultConfig.stub.js) (or no config)? If you are there is no `bg-pink` or `text-red`. Maybe try `bg-pink-500` or `text-red-500`

**Comment by ockam** [comment]
🤦‍♂️ too used to the old Tailwind syntax. Works fine now! Huge thanks for the quick reply!

**Comment by bradlc** [comment]
Hah no worries, I do the same thing when testing Tailwind v1 stuff. `bg-red` was my go-to test class 😅

We should probably throw an error when a class name is unrecognised. Currently unknown class names are just ignored.

**Comment by tobeycodes** [comment]
My workaround for the above is. I did not want to hardcode the color so grab it from the config.

```
import tailwindConfig from '../tailwind.config';

const HeadingInner = styled.h1`
  ${tw`text-5xl font-light tracking-wider uppercase mb-6`}
  color: ${props => (props.dark ? tailwindConfig.theme.colors.gray['600'] : tailwindConfig.theme.colors.primary)}; 
`;
```

**Comment by DevanB** [comment]
> ... the config goes in a [`babel-plugin-macros` config file](https://github.com/kentcdodds/babel-plugin-macros/blob/master/other/docs/user.md#config-experimental). For `styled-components` you would want:
> 
> ```js
> module.exports = {
>   tailwind: {
>     styled: 'styled-components'
>   }
> }
> ```

Would it be possible to specify `styled-components/macro` in the future @bradlc? This would allow us to receive the beautiful `babel-plugin-styled-components` goodies like `displayName`?

The current work-around for this is:

```
const Wrapper = styled.div`
  ${tw`flex flex-col justify-between`}
`;
```

as opposed to
```
const Wrapper = tw.div`flex flex-col justify-between`;
```

**Comment by bradlc** [comment]
@DevanB I’ve looked into this and there’s actually [a bug in `babel-plugin-styled-components`](https://github.com/styled-components/babel-plugin-styled-components/issues/179) that prevents it from working with `tailwind.macro`. The Tailwind macro produces code that looks like this:

```js
import _styled from 'styled-components'

const Wrapper = _styled("div")({
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-between"
});
```

`babel-plugin-styled-components` fails to transform code that looks like this, despite that fact that it is a valid way to create a styled component.

I think we can solve this on our end by updating how the macro transforms the code:

```diff
import _styled from 'styled-components'

- const Wrapper = _styled("div")({
+ const Wrapper = _styled.div({
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-between"
});
```

**Comment by bradlc** [comment]
@DevanB I’ve just published an update (`v1.0.0-alpha.3`) that will hopefully fix the issues with the styled-components babel plugin: `npm install tailwind.macro@next`

I think you might need to have `babel-plugin-macros` before `babel-plugin-styled-components` in your babel config. Let me know if that works ok for you.

**Comment by DevanB** [comment]
@bradlc Updated, tested, and seem to see no differences. **However,** I am using CRA, so didn't specify `babel-plugin-macros` before `babel-plugin-styled-components`, because I'm not overriding the `.babelrc`, yet.

When I create with ``` const Actionbar = styled.div`${tw`flex`}`;```, I get: 

```
const Actionbar = styled_components__WEBPACK_IMPORTED_MODULE_1__["default"].div.withConfig({
  displayName: "App__Actionbar",
  componentId: "sc-1m9o99g-2"
})(["", ""], {
  "display": "flex"
});
```

But when I create with ```const Actionbar = tw.div`flex`;```, I get:

```
const Actionbar = styled_components__WEBPACK_IMPORTED_MODULE_1__["default"].div({
  "display": "flex"
}, p => p.tw);
```

It's quite possible that I need to override the `.babelrc`.

**Comment by bradlc** [comment]
@DevanB Apologies, I misunderstood initally. After looking in to it properly and actually trying it in CRA I _think_ I’ve got it working.

My `babel-plugin-macros.config.js`:

```js
module.exports = {
  tailwind: {
    styled: 'styled-components/macro'
  }
}
```

My component:

```js
import tw from 'tailwind.macro'
let Box = tw.div`uppercase`
```

Transformed code:

```js
let Box = styled_components__WEBPACK_IMPORTED_MODULE_0__["default"].div.withConfig({
  displayName: "App__Box",
  componentId: "sc-1d5tzwi-0"
})({
  "textTransform": "uppercase"
}, p => p.tw);
```

**Comment by DevanB** [comment]
@bradlc looks like that will work. Is this on a remote branch somewhere? I don't mind testing it locally before pushing another alpha.

**Comment by bradlc** [comment]
@DevanB It’s already released under the `next` tag 📦

**Comment by DevanB** [comment]
Interesting, still no luck for me.

<img width="1478" alt="image" src="https://user-images.githubusercontent.com/354652/57203308-ec253500-6f7c-11e9-97b2-3bd47d8372ff.png">

**Comment by bradlc** [comment]
I'm not at my computer right now, but here's a couple of things to check:

1. Are you definitely on the latest version? I think it was `v1.0.0-alpha-5`
2. What order are your `tw` and `styled` imports in, and have you tried swapping them round?

**Comment by DevanB** [comment]
Good check points:

1. Yep, I'm pinned to `alpha.5` for sure.
2. They were not in that order before (since automatic import order puts `styled` before `tw`. But after switching them, it is working!

**Comment by bradlc** [comment]
🎉 Sweet! Unfortunately because of how babel/babel-plugin-macros work if you are importing both `tailwind.macro` and `styled-components/macro` in the same file **you will need `tailwind.macro` first**

**Comment by DevanB** [comment]
Makes complete sense.

The only reason I’m doing that is because on some components I need to conditionally set based on a prop. I know this is open in another issue so once it is doable via the tailwind macro, it shouldn’t be a problem at all!

**Comment by bradlc** [comment]
Where possible I would recommend using `babel-plugin-styled-components` instead of the macro, which would avoid this issue, but I appreciate that CRA doesn’t support custom babel configs

**Comment by DevanB** [comment]
@bradlc does this mean you don't plan to support props like you [stated here](https://github.com/bradlc/babel-plugin-tailwind-components/issues/22#issuecomment-478936715)?

**Comment by apolzon** [comment]
Thanks for getting this working with v1.0.

I'm trying to use this (currently using the alpha.8 release) with cra + @emotion/styled, and the only issue I'm currently running into is the tw literal syntax is not properly outputting the hoisting/sourcemap functionality in dev. (It works fine, just as an inline style)

I read above where the babel-plugin-macros.config.js file needs to be defined with
```js
module.exports = {
  tailwind: {
    styled: 'styled-components/macro'
  }
}
```

however, since I'm using emotion, I tried changing it to:
```js
module.exports = {
  tailwind: {
    styled: '@emotion/styled/macro'
  }
}
```

which still didn't work (I tested by only importing tw and not @emotion/styled/macro in my test file, also with @emotion/styled/macro, both before/after tw).

Definitely not a deal-breaker, I'm fine using the "old" styled component syntax (which is outputting the sourcemap correctly) with interpolated calls to tw for now, but I wanted to post this just in case I had guessed at what the proper config should be incorrectly.

**Comment by ben-foster** [comment]
Thanks for all the work here everyone. Dropping a note here for anyone else who still had the same problem as I did after reading the above responses...

I couldn't figure out how to mix in vanilla CSS in my tailwind styled components with the above solution of writing tailwind styled components this way:
```javascript
import tw from 'tailwind.macro'

const ExampleComponent = tw.div`
  bg-center
  bg-no-repeat
  bg-cover
  h-48
`
```

But I can confirm that if you upgrade tailwind.macro to 1.0.0-alpha.8 add the code mentioned above to your babel-plugin-macros.config.js:
```javascript
module.exports = {
  tailwind: {
    styled: 'styled-components/macro'
  }
}
```
...then you can still write your tailwind styled components this way:
```javascript
import styled from 'styled-components'
import tw from 'tailwind.macro'

const ExampleComponent = styled.div`
  background-image: url('https://picsum.photos/400/240');
  ${tw`
      bg-center
      bg-no-repeat
      bg-cover
      h-48
  `}
  ::after {
     content: "pseudoelement example";
  }
`
```
While I'm here... is there a better way of doing this?

**Comment by bradlc** [comment]
> [...] the tw literal syntax is not properly outputting the hoisting/sourcemap functionality in dev [...]

@apolzon please could you try `tailwind.macro@1.0.0-alpha.9`? In my tests the babel output is the same as using `styled` directly.

**Comment by apolzon** [comment]
I'm still getting the same output as before. My hunch is that it has to do with the babel-plugin-macros.config.js file.

Here's the contents of my config file (located at my client root, its definitely getting read as my server will bomb if I replace '@emotion/styled/macro' with a typo):
```js
module.exports = {
  tailwind: {
    styled: '@emotion/styled/macro'
  }
}
```
(also tried without the /macro suffix)

and here's a snippet from my test page:
```js
import tw from 'tailwind.macro'
import styled from '@emotion/styled/macro'

...

const TailwindHeading = styled.h2`
  ${tw`text-indigo-600`};
`

const NewSyntax = tw.h2`
  text-indigo-600
`

...

<TailwindHeading>
  Tailwind + Emotion Test
</TailwindHeading>
<NewSyntax>
  New Syntax Test
</NewSyntax>
```

And here's what I see in the chrome inspector -- the TailwindHeading styles link to my source code (index.js), while the NewSyntax element is referencing the `<style>` element directly. (I've also tested with the css object style, and that works as well, but I omitted it from this example to try to reduce noise)

<img width="1432" alt="Screen Shot 2019-05-27 at 3 19 41 PM" src="https://user-images.githubusercontent.com/223493/58438189-2b820580-8093-11e9-8f8c-4247231cc5b6.png">

<img width="1439" alt="Screen Shot 2019-05-27 at 3 19 52 PM" src="https://user-images.githubusercontent.com/223493/58438191-3046b980-8093-11e9-92bb-b3a323e42cee.png">

And here is the raw babel output for the two different styled syntaxes (notice the TailwindHeading element has the additional sourceMappingURL defined, while the NewSyntax element does not):

```js
  var TailwindHeading = Object(_emotion_styled_base__WEBPACK_IMPORTED_MODULE_3__["default"])("h2", {
    target: "evcvfon0",
    label: "TailwindHeading"
  })(Object.assign(tailwind_macro_utils_umd_js__WEBPACK_IMPORTED_MODULE_4___default.a.resolveStyle(_tailwindConfig, [{
    "prop": "color",
    "config": "textColor"
  }, {
    "prop": "fontSize",
    "config": "fontSize"
  }], "indigo-600")), ";" + ( false ? undefined : "/*# sourceMappingURL=**omitted** */"));

  var NewSyntax = Object(_emotion_styled_base__WEBPACK_IMPORTED_MODULE_3__["default"])("h2", {
    target: "evcvfon1",
    label: "NewSyntax"
  })(Object.assign(tailwind_macro_utils_umd_js__WEBPACK_IMPORTED_MODULE_4___default.a.resolveStyle(_tailwindConfig, [{
    "prop": "color",
    "config": "textColor"
  }, {
    "prop": "fontSize",
    "config": "fontSize"
  }], "indigo-600")));

  return react__WEBPACK_IMPORTED_MODULE_5___default.a.createElement(react__WEBPACK_IMPORTED_MODULE_5___default.a.Fragment, null, react__WEBPACK_IMPORTED_MODULE_5___default.a.createElement(TailwindHeading, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 54
    },
    __self: this
  }, "Tailwind + Emotion Test"), react__WEBPACK_IMPORTED_MODULE_5___default.a.createElement(NewSyntax, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 57
    },
    __self: this
  }, "New Syntax Test"));
```

I also did a test where I removed the styled.h2`` (TailwindHeading) element and styled import, with the same results -- the returned fragment has the `__source` key, but the initial creation of the element does not have the sourceMappingURL comment.


One last thing that has me puzzled is that the other syntaxes sourcemaps are working fine regardless of whether I define the babel-plugin-macros.config.js file. I'm not sure if this is due to some defaults working out of the box, or if there is something incorrect with my config which is the root issue.

**Comment by ockam** [comment]
@bradlc using alpha.10 with Emotion in Gatsby, everything works fine with my components, but when I build my site, Tailwind defautConfig.stub.js gets lumped into the bundle. SInce it contains ES6 spread operator, this make older browsers crash (Edge, Explorer).

I tried building a simple test case with a new vanilla Gatsby and I don’t have this problem, so, it’s probably not an issue with the macro, but If you have a suggestion on what may cause this in my project, I’m out of idea to try...

**Comment by tobeycodes** [comment]
I am now getting this error `tailwind.macro: Cannot read property 'type' of undefined` when using any class that uses a color e.g. text-red-500, bg-red-900. Other things work like underscore, p-10, m-10, etc. I'm using the following versions if it matters

```
    "styled-components": "^4.3.1",
    "babel-plugin-macros": "^2.6.1",
    "tailwind.macro": "^1.0.0-alpha.10",
    "tailwindcss": "^1.0.4"
```

EDIT: I do not get this anymore after nuking it and starting again.

**Comment by ynnoj** [comment]
Anybody using this with Next.js/SSR?

**Comment by B3Kay** [comment]
I can't get it to work with my tailwind.config.js. I can run it without it. Quite hard to grasp.

**Comment by cuong-nguyen** [comment]
Hi, it seems like I can't do conditional like below
```
<span css={tw`text-xs flex items-center ${true ? 'uppercase' : ''}`}>
```
Is there a way I can make it dynamic like that? Thanks.

**Comment by tobeycodes** [comment]
The type error I mentioned above is back :( 

Works fine for anything like p-10 but colors do not work e.g. bg-red-500

```
./src/components/Button.js: ./src/components/Button.js: tailwind.macro: Cannot read property 'type' of undefined Learn more: https://www.npmjs.com/package/tailwind.macro
  
at assignify (./node_modules/tailwind.macro/macro.js:989:14)
at ./node_modules/tailwind.macro/macro.js:1004:24
at Array.forEach (<anonymous>)
  at assignify (./node_modules/tailwind.macro/macro.js:993:21)
  at getStyles (./node_modules/tailwind.macro/macro.js:1125:11)
  at ./node_modules/tailwind.macro/macro.js:1208:38
  at Array.forEach (<anonymous>)
    at babelPluginMacros.createMacro.configName (./node_modules/tailwind.macro/macro.js:1198:22)
    at macroWrapper (./node_modules/babel-plugin-macros/dist/index.js:61:12)
    at applyMacros **(./node_modules/babel-plugin-macros/dist/index.js:224:14)**
```

**Comment by tobeycodes** [comment]
@bradlc Looks like the error goes away (and colors work) when I comment out this line. Do you have an idea of why this might be happening? Thanks again.

https://github.com/bradlc/babel-plugin-tailwind-components/blob/next/src/getStyles.js#L138

**Comment by cuong-nguyen** [comment]
@bradlc I am on ^1.0.0-alpha.2.
Does it support custom plugin for tailwind? I tried adding sample plugin from https://tailwindcss.com/docs/adding-new-utilities#using-a-plugin but the styles are not available

Extend existing utility seems to work fine
```
module.exports = {
  theme: {
    extend: {
      inset: {
        '4': '1rem',
      },
    },
  },
  variants: {},
  plugins: [
     function({ addUtilities }) {
      const newUtilities = {
        '.rotate-0': {
          transform: 'rotate(0deg)',
        },
        '.rotate-90': {
          transform: 'rotate(90deg)',
        },
        '.rotate-180': {
          transform: 'rotate(180deg)',
        },
        '.rotate-270': {
          transform: 'rotate(270deg)',
        },
      }

      addUtilities(newUtilities, ['responsive', 'hover'])
    }
  ],
}
```

`rotate-` utilities are not there.

**Comment by tobeycodes** [comment]
It does not. I tried the other day using the official custom-forms plugin and it did not work. I opted to use className in that instance.

**Comment by cuong-nguyen** [comment]
@schrapel good to know. Spent quite some time to make this macro works properly then hit this limitation. I think I will go for className approach then.

**Comment by alejandrovasta** [comment]
Hey guys! could you make it work? I have no luck with Gatsby + Tailwind v1.0 and macro :( it runs on dev but when I try to build it does not happen

**Comment by flogy** [comment]
> Hey guys! could you make it work? I have no luck with Gatsby + Tailwind v1.0 and macro :( it runs on dev but when I try to build it does not happen

Hi @alejandrovasta 

I too could not make it work for a long time. Then I discovered this awesome Gatsby theme: [gatsby-theme-tailwindcss](https://github.com/talensjr/gatsby-theme-tailwindcss/blob/master/theme/README.md) 😊

It supports Tailwind 1.x and also works when built (just tried it locally).

**Side note:** the theme also uses `tailwind.macro` etc. internally, so it does not support `custom-forms` at the moment.

Hope this helps!

**Comment by alejandrovasta** [comment]
Puff awesome! it works. Thanks so much

**Comment by zamson** [comment]
Wow this is so much cleaner than using className. So, will tailwind plugins just not work? That's a bummer. The custom-forms plugin is important to me. Are there any plans to make this work?

**Comment by leonardoelias** [comment]
Wooow, this fantastic.

Any examples using next.js?

**Comment by tgrecojs** [comment]
@leonardoelias the next.js examples repository has one that incorporates tailwind with emotion-js. not sure if you already came across that, but if not here it is: [next.js examples - withtailwindcss-emotion](https://github.com/zeit/next.js/tree/canary/examples/with-tailwindcss-emotion) 😎👍

**Comment by tobeycodes** [comment]
I was using this with Next.js but now I pulled it out and just use className. Unfortunately I kept running into issues with this repo. I want to spend some time soon to try fix the issues I identified and add full tailwind v1 compatability

**Comment by deadcoder0904** [comment]
@schrapel same. a project like this is better in the main repo where it gets maintained with recent changes or else we won't be able to enjoy new features :)

**Comment by tobeycodes** [comment]
I don't think it will ever go into the main repo. I don't think it is something @admwathan would want.

I hope that @bradlc will find some time to review the PRs I will add but I understand that OSS is hard and we don't always have as much time to devote as we would like.

**Comment by noliiva** [comment]
Hello there.
I'm pretty new to tailwindcss and I try to use it with the macro (alpha-10).
Base styles works fine but I can't use "container" class. I get this error: `Error: Couldn’t resolve Tailwind class name: container`.
I try to addComponent in config file to replace "container", but this didn't work.
Did I miss something ?

**Comment by tobeycodes** [comment]
@noliiva the container class is not supported. You should use `className` for that

**Comment by noliiva** [comment]
Oh, ok. That's sad. :p
Are there any plans to support this feature someday?

**Comment by mergims** [comment]
@noliiva just had the same problem. I changed macro.js on row 1108 and it worked.

From:
`var obj$1 = state.isProd ? resolveStyleFromPlugins(state.default, className) : ( obj$3 = {}, obj$3['__spread__' + index] = state.tailwindUtilsIdentifier.name + '.resolveStyleFromPlugins(' + state.tailwindConfigIdentifier.name + ', "' + className + '")', obj$3 );`

To:
`var obj$1 = state.isProd ? resolveStyleFromPlugins(state.config.default, className) : ( obj$3 = {}, obj$3['__spread__' + index] = state.tailwindUtilsIdentifier.name + '.resolveStyleFromPlugins(' + state.tailwindConfigIdentifier.name + ', "' + className + '")', obj$3 );`

Basically, just change the config reference from `state.default` to `state.config.default` in the resolveStyleFromPlugins function call.

**Comment by noliiva** [comment]
Oh, that's interesting !
Perhaps this should be added officially to the macro. :)

**Comment by mergims** [comment]
@noliiva Yep. An option would be to fork the `next` branch and implement the fix if you need it urgently. And then add the git repo as a dependency in your project.

**Comment by tobeycodes** [comment]
This might be useful. In a new project we decided to not use this package. Instead we load the Tailwind CSS file (purgecss/cssnano in production) and use a classnames function to choose the classes. It offers a similar developer experience. With Tailwind css v2 just around the corner this package we chose not to rely on this package for the time being. 

Hope this helps someone

https://github.com/JedWatson/classnames

```
// Button.tsx
const Button: FC<ButtonProps> = ({ children, className, variant, size }: ButtonProps) => (
  <button
    className={classNames({
      'font-bold text-white': true,
      // Variants
      'border border-white': !variant,
      'bg-orange-500': variant === 'primary',
      'bg-gradient-brand': variant === 'secondary',
      // Sizes
      'px-5 rounded-3xl h-10': !size,
      'px-4 h-8 text-xs rounded-2xl md:px-5 md:text-base md:rounded-3xl md:h-10': size === 'small'
    })}
  >
    {children}
  </button>
);
```
